core/src/main/java/org/elasticsearch/index/IndexModule.java

```diff
@@ -359,6 +360,12 @@ public IndexService newIndexService(NodeEnvironment environment, IndexService.Sh
                 searchOperationListeners, indexOperationListeners);
     }
 
+    public MapperService newIndexMapperService(MapperRegistry mapperRegistry) throws IOException {
```

Comment:
Can you add documentation that this mapper service may only be used fo administrative purposes, and not eg. actuallly parsing documents?